### DISCUSSION TITLE
Email Auth duplicate key value violates unique constraint

### DISCUSSION THREAD

**Discussion by Tristannos** [serverpod/serverpod#2738] [Q&A] [answered]
When a new user is trying to register I get this error in the serverpod_session_log:
 `DatabaseException: duplicate key value violates unique constraint "serverpod_user_info_pkey"`
 
 Even Sign in with Apple. If it's an new account it gives the same error.

```
#0      DatabaseConnection._query (package:serverpod/src/database/adapters/postgres/database_connection.dart:404:9)
<asynchronous suspension>
#1      DatabaseConnection._mappedResultsQuery (package:serverpod/src/database/adapters/postgres/database_connection.dart:458:18)
<asynchronous suspension>
#2      DatabaseConnection.insert (package:serverpod/src/database/adapters/postgres/database_connection.dart:143:9)
<asynchronous suspension>
#3      DatabaseConnection.insertRow (package:serverpod/src/database/adapters/postgres/database_connection.dart:156:18)
<asynchronous suspension>
#4      Users.createUser (package:serverpod_auth_server/src/business/users.dart:28:16)
<asynchronous suspension>
#5      Emails.createUser (package:serverpod_auth_server/src/business/email_auth.dart:178:18)
<asynchronous suspension>
#6      EmailEndpoint.createAccount (package:serverpod_auth_server/src/endpoints/email_endpoint.dart:74:12)
<asynchronous suspension>
#7      EndpointDispatch.handleUriCall (package:serverpod/src/server/endpoint_dispatch.dart:127:20)
<asynchronous suspension>
#8      Server._handleRequest (package:serverpod/src/server/server.dart:296:18)
<asynchronous suspension>
```

**To Reproduce**
1. Register users
2. Export the tables `serverpod_user_info`, `serverpod_email_auth`
3. And import them back in again
4. The records are precisely the same as before only now.
5. Register a user trough the same commands as shown in de Docs.
6. When trying to verify the connection code it gives this error.

I am trying to make an account for "test@gmail.com". But as you see there is no record
**Screenshots**
![image](https://github.com/user-attachments/assets/9d615662-41bc-45df-8465-94bc7034ec27)

![image](https://github.com/user-attachments/assets/19bd16b9-ace7-43f6-a5bb-6616ab3972dc)

![image](https://github.com/user-attachments/assets/434c5582-1742-4bf5-906f-590698642a2d)
But as you see it does not already exist.

**Serverpod versions**
 -  _CLI Version: 2.0.2

**Platform information**
Using an iPhone.

**Comment by SandPod** [comment, marked as answer]
Hey @Tristannos 👋 

### The problem
The identity column in Serverpod is generated using the `Serial` keyword in Postgres along with the `PRIMARY KEY` keyword.

The `Serial` keyword is simply an auto incrementer, giving each record a new value.
The `PRIMARY KEY` adds an index that ensures that two rows never have the same identity value.

When you export you data and import it to a new database the `Serial` auto-incrementer might be reset, meaning that you will start creating new rows from `1` instead of the number of records that already exist, resulting in this collision.

### Solution
You need to recreate the sequence when dumping the table.

There is some information to read about it on the `pg_dump` command here: https://www.postgresql.org/docs/current/app-pgdump.html (search for sequence).

There are also a lot of resource online for how to preserve a sequence.

Here are two example of stackoverflow thread on the subject: https://stackoverflow.com/questions/55485135/in-postgresql-after-importing-the-csv-file-the-serial-number-is-not-synchroniz, https://stackoverflow.com/questions/33033642/pg-dump-setting-of-sequences

**Reply by Tristannos** [reply]
Thanks for the quick response! Helped me a lot!
